registrations/serializers.py

```diff
@@ -61,6 +49,68 @@ def create(self, validated_data):
         else:
             raise DRFPermissionDenied("The waiting list is already full")
 
+    def validate(self, data):
+        errors = {}
+        registration = data["registration"]
+        mandatory_fields = {mf.id for mf in registration.mandatory_fields.all()}
+
+        mandatory_field_mapping = [
+            {
+                "field_ids": ["street_address"],
+                "mandatory_field_id": MandatoryField.DefaultMandatoryField.ADDRESS,
+            },
+            {
+                "field_ids": ["city", "zipcode"],
+                "mandatory_field_id": MandatoryField.DefaultMandatoryField.CITY,
+            },
+            {
+                "field_ids": ["name"],
+                "mandatory_field_id": MandatoryField.DefaultMandatoryField.NAME,
+            },
+            {
+                "field_ids": ["phone_number"],
+                "mandatory_field_id": MandatoryField.DefaultMandatoryField.PHONE_NUMBER,
+            },
+        ]
+
+        for m in mandatory_field_mapping:
+            if m["mandatory_field_id"] in mandatory_fields:
+                for field in m["field_ids"]:
+                    val = data.get(field)
+                    if not val:
+                        errors[field] = _("This field must be specified.")
```

Comment:
As far as I can tell, if there's a non-default mandatory field with an ID `foobar` this validation logic wouldn't be able to catch it as an error since it's only iterating through `mandatory_field_mapping` .